experiments/AMIP/modular/components/land/bucket_init.jl

```diff
@@ -49,77 +51,87 @@ multiple dispatch on `surface_fluxes`.
 struct CoupledAtmosphere{FT} <: AbstractAtmosphericDrivers{FT} end
 
 """
-    surface_fluxes(Y, p, 
-                    t::FT,
-                    parameters::P,
-                    atmos::PA,
-                    ) where {FT <: AbstractFloat, P <: BucketModelParameters{FT},  PA <: CoupledAtmosphere{FT}}
+    surface_fluxes(atmos::CoupledAtmosphere{FT},
+                    model::BucketModel{FT},
+                    Y,
+                    p,
+                    ) where {FT <: AbstractFloat}
 
 Computes the turbulent surface fluxes terms at the ground for a coupled simulation.
 """
 function ClimaLSM.Bucket.surface_fluxes(
-    Y::ClimaCore.Fields.FieldVector,
-    p::ClimaCore.Fields.FieldVector,
-    t::FT,
-    parameters::BucketModelParameters{FT, P},
     atmos::CoupledAtmosphere{FT},
-) where {FT <: AbstractFloat, P}
-    # coupler has done its thing behind the scenes already
+    model::BucketModel{FT},
+    Y,
+    p,
+    _...,
+) where {FT <: AbstractFloat}
     return (turbulent_energy_flux = p.bucket.turbulent_energy_flux, evaporation = p.bucket.evaporation)
```

Comment:
It ends up being helpful for us to split the energy flux into LHF and SHF. however, I know the atmosphere only computes the sum. let's discuss a bit more!

Reply:
Do you mean ClimaLSM splits the energy flux into LHF and SHF?

Reply:
Hm, yes - so when we compute it ourselves, we call surface fluxes, which returns the LHF and SHF separately. then we add them together outside of `surface_fluxes()`. 
https://github.com/CliMA/ClimaLSM.jl/blob/507ec51e4032593b1cbe3ff10e916d8ee9f06e39/src/Bucket/Bucket.jl#L473-L475

here is the prescribedAtmos version:
https://github.com/CliMA/ClimaLSM.jl/blob/507ec51e4032593b1cbe3ff10e916d8ee9f06e39/src/SharedUtilities/drivers.jl#L225

This is hacky, but we could do something like
`return (lhf = FT(0), shf = p.bucket.turbulent_energy_flux, vapor_flux = p.bucket.evaporation, Ch = FT(0)` here?

[We ended up returning LHF and SHF separately, as well as passing back Ch, because we are trying to adjust for canopy resistance/stomatal conductance in a new PR, but in an offline fashion, since SF.jl doesnt account for it yet.

Reply:
@LenkaNovak what do you think?

Reply:
Hmm, @kmdeck am I interpreting this correctly in that this function output is not being used by the simple bucket? I.e., the fluxes calculated by the coupler are only feeding into atmos right now, while this is just a placeholder for the canopy? If so, @juliasloan25 I would comment on this in the code, and do what Kat suggests (though perhaps setting all the output fields with unitialized arrays - using `similar()`).

Reply:
What is `Ch` here?

Reply:
That's the transfer coefficient (comes to out SF.jl)

Reply:
@kmdeck would it be possible to add `lhf` and `shf` to the bucket's cache so that we can access each component individually?

Reply:
We could do that, but as I recall the atmosphere only returns the sum of them, so Im not sure it would materially change things? let's discuss when we chat at 10

Reply:
> Hmm, @kmdeck am I interpreting this correctly in that this function output is not being used by the simple bucket? I.e., the fluxes calculated by the coupler are only feeding into atmos right now, while this is just a placeholder for the canopy? If so, @juliasloan25 I would comment on this in the code, and do what Kat suggests (though perhaps setting all the output fields with unitialized arrays - using `similar()`).

Hey Lenka,

in the coupled sim, the fluxes computed by the coupler or the atmosphere are needed by the bucket model. But the bucket model needs only the net energy flux, not split into LHF and SHF.

In the long run, Im not sure if the canopy model really needs it split up either (though Im sure it would be useful for diagnostics and testing, so not bad to have). But, since SurfaceFluxes.jl does not take into account surface conductance (stomatal conductance or soil conductance), we returned them seperately so that as best as we could, we could account for them in standalone runs. 

That is shown here
https://github.com/CliMA/ClimaLSM.jl/blob/0426df82cca3aaba0c8e8d9034b7183a685f6886/src/Vegetation/Canopy.jl#L250-L268

But, E also appears in the SHF, so this still isnt really correct. really just a stand-in on our end.

happy to chat more!

Reply:
Hmm, the atmos interface for this has changed quite a bit recently. If I'm interpreting this correctly, they now store `surface_conditions` and then set them [here](https://github.com/CliMA/ClimaAtmos.jl/blob/248e0f95bc27573494704da0bf670289c15fce08/src/tendencies/vertical_diffusion_boundary_layer.jl#L307), with `lhf` and `shf` as separate components. We use this function (called at a different time) in the coupler. If it's easier for the Bucket to keep the sum then that's fine too. We can write an adapter for it. Let us know, Kat, what you decide to do and we'll adapt.

Reply:
Kat and I discussed this offline. We decided it will be helpful in the future for the Bucket to have `lhf` and `shf` as separate components. Kat will work on making this change in ClimaLSM and let us know when it's done so we can store `lhf` and `shf` instead of setting one to 0 and the other storing the sum. We agreed that this should be a separate PR :)